aws/terminator.yml

```diff
@@ -39,100 +50,100 @@
             Principal:
               Service: "lambda.amazonaws.com"
             Effect: "Allow"
+
     - name: create iam policy for terminator functions
       tags: iam
       iam_policy:
         iam_type: role
-        iam_name: "{{ api_name }}-terminator-{{ stage }}"
-        policy_name: "{{ api_name }}-terminator-{{ stage }}"
+        iam_name: "{{ iam_role_name }}"
+        policy_name: "{{ iam_policy_name }}"
         state: present
         policy_json: "{{ terminator_policy | to_nice_json }}"
+
     - name: create virtualenv with terminator requirements
       tags: lambda
       pip:
-        requirements: "{{ playbook_dir }}/requirements.txt"
+        requirements: "{{ virtualenv_requirement_file }}"
         virtualenv: "{{ packaging_dir }}/terminator-requirements/python"
-        virtualenv_python: python3.9
+        virtualenv_python: "{{ lambda_python_version }}"
+      check_mode: false
+
     - name: package terminator requirements
       tags: lambda
-      lambda_package:
-        src: "{{ packaging_dir }}/terminator-requirements"
+      archive:
```

Comment:
This isn't a full replacement for the `lambda_package` module. The generated archive changes based on the timestamps of the included files, which results in a new Lambda version being created even when the file contents are unchanged.

Reply:
Fixed.
Idempotency check for the lambda layer archive is assumed by downloading the one stored in AWS and compare it with the local one in a deterministic way (setting a constant time value)

Reply:
Why add a custom module to do the comparison instead of using the existing module to create the archive?

Reply:
existing I did not found a way in the module to create archive in a deterministic way, so we will always have a diff between the sha from aws and the local one, please suggest if there is a way to solve that

Reply:
Sorry, I wasn't specific enough. I was asking why replace `mattclay.aws.lambda_package`?

Reply:
`mattclay.aws.lambda_package` works with `mattclay.aws.lambda_layer` because the lambda_layer from that collection expects an archive with a deterministic timestamp to ensure idempotency, this behavior is not part of the `amazon.aws` collection. However I can keep `amazon.aws.lambda_package` and perform the logic on the playbook, this is almost what I have done but without the dependency of the `mattclay.aws` collection

Reply:
What's the reason for wanting to remove the dependency on the `mattclay.aws` collection to begin with?

Reply:
> I think the intend was to migrate to a collection supported by the cloud team. @jillr could you please clarify

This was never the intent. There was a need to update `community.aws`, in particular migrating to the newer `iam_role` module. There's never been a problem with the `mattclay.aws` collection. If there are modules from the supported `amazon.aws` collection that can fully replace modules from the `mattclay.aws` collection, we can consider those, but that is not apparently the case here with the `lambda_package`.

Reply:
ok then I will keep lambda_package from `mattclay.aws`

Reply:
FYI - I've updated the `mattclay.aws` collection to version 3.0.0. It should now play nice with `amazon.aws` 5.x.